device/cuda/include/cuda/seeding/seed_finding.hpp

```diff
@@ -55,7 +55,7 @@ struct seed_finding {
     /// Callable operator for the cuda seed finding
     ///
     /// @return seed_container is the vecmem seed container
-    host_seed_container operator()(
+    host_seed_container& operator()(
```

Comment:
Wait... what? Are you returning a permanent object all of a sudden? This does not seem thread-safe at all... :thinking:

Reply:
The reason for this change was to use the return output of certain algorithm as an input of next algorithm directly. With copy method, the memory resource is missing so we cannot use it directly for the next algorithm.
```
auto output = cuda_algorithm(input) // output cannot be used as an input for the next algorithm bec
``` 

But as you mentioned that it is not thread-safe, I reverted it to copy method and do some modification:
```
algorithm::output_type output(mr);
output = cuda_algorithm(input)
```
Is it okay to you?

Reply:
Sure. (Sorry for the late answer... :frowning:)